base_locale_uom_default/models/res_lang.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ResLang(models.Model):
+
+    _inherit = 'res.lang'
+
+    default_uom_ids = fields.Many2many(
+        string='Default Units',
+        comodel_name='product.uom',
+    )
+
+    @api.multi
+    @api.constrains('default_uom_ids')
+    def _check_default_uom_ids(self):
+        for record in self:
+            uoms = {}
+            for uom in record.default_uom_ids:
+                try:
+                    raise ValidationError(_(
+                        'Only one default unit of measure per category may '
+                        'be selected, there is more than one in the %s '
+                        'category.'
+                    ) % (
+                        uoms[uom.category_id.id].name,
+                    ))
+                except KeyError:
+                    uoms[uom.category_id.id] = uom.category_id
+
+    @api.model
+    def default_uom_by_category(self, category_name, lang=None):
+        """Return the default UoM for language for the input UoM Category.
+
+        Args:
+            category_name (str): Name of the UoM category to get the default
+            for.
+            lang (ResLang or str, optional): Recordset or code of the language
+            to get the default for. Will use the current user language if
+            omitted.
+
+        Returns:
+            ProductUom: Unit of measure representing the default, if set.
+            Empty recordset otherwise.
+        """
+        if lang is None:
+            lang = self.env.user.lang
+        if isinstance(lang, basestring):
+            lang = self.env['res.lang'].search([('code', '=', lang)])[:1]
```

Comment:
this could use ``limit=1`` instead of slicing and you could use ``('category_id.name`,'=', category_name)``